src/ripple/basics/impl/StringUtilities.cpp

```diff
@@ -31,14 +32,11 @@
 
 namespace ripple {
 
-// NIKB NOTE: This function is only used by strUnHex (std::string const& strSrc)
-// which results in a pointless copy from std::string into std::vector. Should
-// we just scrap this function altogether?
-int strUnHex (std::string& strDst, std::string const& strSrc)
+std::pair<Blob, bool> strUnHex (std::string const& strSrc)
```

Comment:
Should we just scrap this function (and its encoder) and go with `boost::hex` and `boost::unhex`? Here's the [documentation](http://www.boost.org/doc/libs/1_61_0/libs/algorithm/doc/html/the_boost_algorithm_library/Misc/hex.html), but it's a header-only library which we already use and reduces the code that we need to write.